.cbmc-batch/jobs/aws_mul_size_checked/aws_mul_size_checked_harness.c

```diff
@@ -0,0 +1,46 @@
+#include <aws/common/math.h>
+#include <proof_helpers.h>
+
+/**
+ * Multiplies a * b and returns the result in *r. If the result
+ * overflows, returns AWS_OP_ERR; otherwise returns AWS_OP_SUCCESS.
+ */
+AWS_STATIC_IMPL int __aws_mul_u64_checked(uint64_t a, uint64_t b, uint64_t *r) {
+    uint64_t x = a * b;
+    *r = x;
+    if (a != 0 && (a > UINT32_MAX || b > UINT32_MAX) && x / a != b) {
+        return aws_raise_error(AWS_ERROR_OVERFLOW_DETECTED);
+    }
+    return AWS_OP_SUCCESS;
+}
+
+/**
+ * Multiplies a * b and returns the result in *r. If the result
+ * overflows, returns AWS_OP_ERR; otherwise returns AWS_OP_SUCCESS.
+ */
+AWS_STATIC_IMPL int __aws_mul_u32_checked(uint32_t a, uint32_t b, uint32_t *r) {
+    uint32_t x = a * b;
+    *r = x;
+    if (a != 0 && (a > UINT16_MAX || b > UINT16_MAX) && x / a != b) {
+        return aws_raise_error(AWS_ERROR_OVERFLOW_DETECTED);
+    }
+    return AWS_OP_SUCCESS;
+}
+
+void aws_mul_size_checked_harness() {
+    size_t r = nondet_size_t();
+    size_t __r = nondet_size_t();
+    __CPROVER_assume(r == __r);
+
+    if(nondet_int()){
```

Comment:
Maybe add a comment that doing it with the full range of undet inputs leads to excessively long runtimes?